src/transformers/models/auto/auto_factory.py

```diff
@@ -517,6 +542,33 @@ def register(cls, config_class, model_class, exist_ok=False):
             )
         cls._model_mapping.register(config_class, model_class, exist_ok=exist_ok)
 
+    @classmethod
+    def _check_and_return_if_adapter_model(
+        cls,
+        model_id: str,
+        revision: str = None,
+        use_auth_token: Optional[str] = None,
+        commit_hash: Optional[str] = None,
+    ) -> bool:
+        r"""
+        Simply checks if the model stored on the Hub or locally is an adapter model or not
+        """
+        adapter_cached_filename = None
+        try:
```

Comment:
You can avoid the try except by setting the private args of `cached_file` to not raise exceptions.

Reply:
Should be fixed in 032077a